### ISSUE TITLE
Handling of -Bsymbolic-function in CMakeLists.txt is deficient

### ISSUE THREAD

**Issue by hawicz** [json-c/json-c#569]
The old autoconf machinery had an actual link test to make sure that the -Bsymbolic-function complier argument actually worked.  CMakeLists.txt is missing this, and thus building on systems that don't support that, such as osx, fails unless -DDISABLE_BSYMBOLIC is passed to cmake.

Once this is fixed, the special case in .travis.yml could be cleaned up as well.

**Comment by dota17** [comment]
like this?
```
set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-Bsymbolic -Wl,-Bsymbolic-functions")
set (CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -Wl,-Bsymbolic -Wl,-Bsymbolic-functions")
```
https://github.com/apitrace/apitrace/blob/master/wrappers/CMakeLists.txt

**Comment by hawicz** [comment]
No, we already set -Bsymbolic-functions, the problem is that we do it unconditionally without checking whether it is supported by the linker.

**Comment by hawicz** [comment]
Fixed by @besser82 in PR #572
